File: problems/0001/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #First solution
        # myDict = {}
        # diffs = [target - n for n in nums]
        # for idx,n in enumerate(nums):
        #     myDict[n] = idx
        # for idx, diff in enumerate(diffs):
        #     if diff in myDict and myDict[diff] != idx:
        #         return [idx,myDict[diff]]

        #Second solution
        myDict = {}
        for idx, n in enumerate(nums):
            diff = target - n
            if diff in myDict:
                return [myDict[diff], idx]
            myDict[n] = idx

if __name__ == "__main__":
    print(Solution().twoSum([3,2,4],6))

File: problems/0001/solution.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> myMap;
        int diff = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            diff = target - nums[i];
            if (myMap.find(diff) != myMap.end())
            {
                return vector<int>{myMap[diff], i};
            }
            myMap[nums[i]] = i;
        }
        return vector<int>{0, 0};
    }
};

int main()
{
    Solution sol;
    vector<int> nums = {2, 6, 11, 15};
    for (const auto &n : sol.twoSum(nums, 8))
    {
        cout << n;
    }
}